### PR TITLE
Updated L1T Fine Grain Bit Definition with bugfixes

### PR DESCRIPTION
**Pull request by mulhearn**
This implements new Fine Grain Bit logic that better matches the hardware.

### CODE REVIEW

**Comment by mulhearn** [comment]
Past iterations have complained about the long if-else trees.  @mnorthup, can fix those in this branch on cms-l1t-offline.  Just work relative to version "CMSSW_7_6_0" so we won't need to rebase again, then push your changes directly to this branch, updating the PR.

**Comment by mnorthup** [comment]
I have no idea what you are talking about. Maybe you intended to refer to
someone else?

**Comment by mnorthup6690** [comment]
So I cleaned up the ieta and iphi conversion functions in HcalTriggerPrimitiveAlgo.cc. I used a big switch statement for one of the functions because I don't know how else to clean it up.

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
@mnorthup6690 is this version tested as thoroughly as the previous?  I'm trying to decide if I should just make a PR with these updates for 75X or continuing to push to use your original PR...

**Comment by mnorthup6690** [comment]
I would say that this version is not tested as thoroughly as the previous, though I would think the results would be the same.

**Comment by mnorthup6690** [comment]
Just to be safe, why don't we keep the 75X PR the same and make this PR for 80X only.

**Comment by mulhearn** [comment]
+1

File: SimCalorimetry/HcalTrigPrimProducers/src/HcalTrigPrimDigiProducer.cc [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include "CondFormats/HcalObjects/interface/HcalElectronicsMap.h"
 #include "CondFormats/HcalObjects/interface/HcalLutMetadata.h"
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
-
+#include "TH2.h"
```
Comment (davidlange6) [inline SimCalorimetry/HcalTrigPrimProducers/src/HcalTrigPrimDigiProducer.cc]: @mulhearn -why is this needed?

Comment (mnorthup) [inline SimCalorimetry/HcalTrigPrimProducers/src/HcalTrigPrimDigiProducer.cc]: remove

**Comment by mnorthup6690** [comment]
I think someone sent permission to the wrong Michael Northup. Could someone remove the permission of mnorthup. Thanks.

**Comment by mulhearn** [comment]
He doesn't have any special permissions, the poor guy is just getting annoying messages from us by mistake...

**Comment by mnorthup** [comment]
Somehow my account (mnorthup) got added to the cms-sw/cmssw thread. Can you
remove it for me?

**Comment by mulhearn** [comment]
@mnorthup very sorry for inadvertently adding you.  I unfortunately don't have permission to remove you from the link.  I think the easiest way out for you is to simply follow the link at the bottom of the e-mail that says "view it on github" then click the "Unsubscribe" button on that page.

**Comment by smuzaffar** [comment]
mnorthup, please click on the "Unsubscribe" button on the left side of this Pull reuqest to not receive any more messages.

**Comment by mulhearn** [comment]
@mnorthup6690 can you remove the (I believe unneeded) TH2F.h include as noticed by David above...

File: SimCalorimetry/HcalTrigPrimProducers/src/HcalTrigPrimDigiProducer.cc (hunk 1 above)
Comment (mulhearn) [inline SimCalorimetry/HcalTrigPrimProducers/src/HcalTrigPrimDigiProducer.cc]: @davidlange6 that must be a mistake.  I've asked Michael to remove it and update the pr.
